### PR TITLE
Fixes and updates

### PR DESCRIPTION
**Pull request by uguraslan**
- As a temporary solution to use `msdf-generator` as a library in other projects, pushed `dist` folder with generated code.
- instead of running msdf-bmfont as an CLI command inside `genFont.ts`, now it uses `generateBMFont` from `msdf-bmfont-xml` package directly. This solves some edge issues related to path resolution of `msdf-bmfont` when there is a complex dependency tree.
- As a temporary solution, instead of using the npm package for `msdf-bmfont-xml`, it uses direct GitHub URL with a fixed commit hash. Because the package maintainers added more binaries for Linux arm64 and MacOS arm64 platforms. But they haven't created a release for it for the last 10 months.
- now the project requires Node 18+ engine.

### CODE REVIEW

**Review by frank-weindel — CHANGES_REQUESTED** [review]
Great work! LGTM in principal. Though the `dist` directory should be removed from the git repo and only published to NPM.

File: package.json [hunk 1]
```diff
@@ -15,17 +15,19 @@
   "keywords": [],
   "author": "",
   "license": "ISC",
```
Comment (michielvandergeest) [inline package.json:17]: License mismatch with the actual license of the package (Apache 2.0)

**Review by michielvandergeest — CHANGES_REQUESTED** [review]

**Comment by uguraslan** [comment]
I removed the `dist` folder and added `prepublishOnly` script in order to build the project before npm publish. Also updated license info and added license text to all files, if possible.

**Review by michielvandergeest — APPROVED** [review]
Looks good to me now with these last changes!

**Review by frank-weindel — APPROVED** [review]

**Pull request merged**